### PR TITLE
Enable unit and integration tests on FreeBSD

### PR DESCRIPTION
**Pull request by tstruk**
* Add include of system endian header file
* configure.ac: add a socket tool check specific to FreeBSD
* tests: make tests more portable

### CODE REVIEW

**Comment by flihp** [comment]
not sure what "main" is, how it got enabled or why it's failing: some help @tstruk ?

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+task:
+  env:
+    CFLAGS: -I/usr/local/include
+    LDFLAGS: -L/usr/local/lib
+    ibmtpm_name: ibmtpm1119
```
Comment (flihp) [inline .cirrus.yml]: Any reason we can't use the same version of the simulator as the travis-ci config?

Comment (tstruk) [reply .cirrus.yml]: That's changed to align with travis.

**Comment by tstruk** [comment]
> Looks like the new service added in this commit isn't working properly. Some help @tstruk ?

Sorry for the delay. It took me some time to debug this. RM couldn't register it's name on the session bus, but there wasn't any errors in the logs. I had to add --allow-root option in the int-test-setup.sh and it is working fine now.

**Comment by tstruk** [comment]
This PR needs to be merged to unblock this tools PR: https://github.com/tpm2-software/tpm2-tools/pull/1996

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - pkg upgrade -y
+    - pkg install -y bash gmake coreutils libtool pkgconf autoconf autoconf-archive openssl libgcrypt
+    - pkg install -y automake glib expat dbus dbus-glib cmocka wget git gettext-runtime tpm2-tss
+    - wget --quiet --show-progress --progress=dot:giga "https://downloads.sourceforge.net/project/ibmswtpm2/$ibmtpm_name.tar.gz"
+    - shasum -a256 $ibmtpm_name.tar.gz | grep ^fc3a17f8315c1f47670764f2384943afc0d3ba1e9a0422dacb08d455733bd1e9
+    - mkdir -p $ibmtpm_name
+    - tar xvf $ibmtpm_name.tar.gz -C $ibmtpm_name && cd $ibmtpm_name/src
+    - sed -i '' -e 's/gcc/clang/g' makefile
+    - sed -i '' -e 's/-Wall //g' makefile
+    - sed -i '' -e 's/-Werror //g' makefile
```
Comment (williamcroberts) [inline .cirrus.yml:21]: fun

**Pull request merged**